torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6310,9 +6316,23 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            supports_forward_ad=True,
            sample_inputs_func=sample_inputs_max_min_binary,),
+    # `softmax` supports different dtypes based on whether `dtype` argument,
+    # is passed or not. Hence two OpInfo entries, one with dtype and other without.
+    OpInfo('nn.functional.softmax',
+           aliases=('softmax',),
+           aten_name="softmax",
+           dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
+           sample_inputs_func=sample_inputs_softmax_variant,
+           supports_out=False),
+    OpInfo('nn.functional.softmax',
+           aliases=('softmax',),
+           variant_test_name="with_dtype",
+           aten_name="softmax",
+           dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           sample_inputs_func=partial(sample_inputs_softmax_variant, with_dtype=True),
+           supports_out=False),
     OpInfo('nn.functional.hardswish',
            aten_name="hardswish",
-           supports_autograd=True,
```

Comment:
Parameters similar to: `supports_autograd, supports_gradgrad, assert_autodiffed, aliases` are removed when they are set to the default values.

Reply:
I vote against removing this here, since I like to keep the scope of the PR narrow. Not saying is not valid, but this change should be done in a follow-up PR. 

Imagine a scenario where one of the unrelated changes breaks something and needs to be reverted. Now the main functionality of the PR also gets reverted although it was working fine.

Reply:
Sure, this makes sense. Thanks @pmeier !